src/core/server/services/comments/stats.ts

```diff
@@ -0,0 +1,134 @@
+import { Redis } from "ioredis";
+
+import { User } from "coral-server/models/user";
+import {
+  retrieveDailyTotal,
+  retrieveHourlyTotals,
+  updateDailyCount,
+  updateHourlyCount,
+} from "coral-server/services/stats/helpers";
+
+import { GQLUSER_ROLE } from "coral-server/graph/schema/__generated__/types";
+
+export interface DailyCommentCounts {
+  total: number;
+  staff: number;
+}
+
+function dailyCommentCountKey(tenantID: string, siteID: string, today: number) {
+  return `stats:${tenantID}:${siteID}:dailyCommentCount:${today}`;
+}
+
+function hourlyCommentCountKey(tenantID: string, siteID: string, hour: number) {
+  return `stats:${tenantID}:${siteID}:hourlyCommentCount:${hour}`;
+}
+
+function dailyStaffCommentCountKey(
+  tenantID: string,
+  siteID: string,
+  today: number
+) {
+  return `stats:${tenantID}:${siteID}:dailyStaffCommentCount:${today}`;
+}
+
+function hourlyStaffCommentCountKey(
+  tenantID: string,
+  siteID: string,
+  hour: number
+) {
+  return `stats:${tenantID}:${siteID}:hourlyStaffCommentCount:${hour}`;
+}
+
+export async function updateCommentTotals(
+  redis: Redis,
+  tenantID: string,
+  siteID: string,
+  now: Date
+) {
+  await updateHourlyCount(redis, tenantID, siteID, now, hourlyCommentCountKey);
+  await updateDailyCount(redis, tenantID, siteID, now, dailyCommentCountKey);
```

Comment:
I'd wrap these into the same redis operation instead, using `MULTI` to prevent the back/forth with redis.